power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -163,38 +178,506 @@ template <bool sym> class NewtonRaphsonSESolver {
+        auto const f_x_complex_abs_ui_inv_row = dot(u_state.abs_ui_inv, f_x_complex_row);
+        auto const injection_jac_diagonal = jacobian_diagonal_component(f_x_complex_abs_ui_inv_row, f_x_complex_row);
+        add_injection_jacobian(diag_block, injection_jac_diagonal);
+        rhs_block.tau_p() -= real(f_x_complex_row);
+        rhs_block.tau_q() -= imag(f_x_complex_row);
+    }
+
+    /// @brief Adds contribution of G(i, i) form the shunt.
+    /// Note: The sign of Y_s is inverted per injection direction.
+    ///
+    /// @param block LHS(i, i)
+    /// @param rhs_block RHS(i)
+    /// @param ys shunt admittance related to the shunt measurement
+    /// @param u_state Voltage state of iteration Voltage state of iteration
+    /// @param measured_power measured shunt power
+    void process_shunt_measurement(NRSEGainBlock<sym>& block, NRSERhs<sym>& rhs_block, auto const& ys,
+                                   auto const& u_state, auto const& measured_power) {
+        auto const hm_ui_ui_ys = hm_complex_form(-ys, u_state.ui_ui_conj);
+        auto const nl_ui_ui_ys = dot(hm_ui_ui_ys, u_state.abs_ui_inv);
+        auto const f_x_complex = sum_row(hm_ui_ui_ys);
+        auto const f_x_complex_abs_ui_inv = sum_row(nl_ui_ui_ys);
+
+        auto jac_block = calculate_jacobian(hm_ui_ui_ys, nl_ui_ui_ys);
+        jac_block += jacobian_diagonal_component(f_x_complex_abs_ui_inv, f_x_complex);
+        auto const& block_F_T_k_w = transpose_multiply_weight(jac_block, measured_power);
+        multiply_add_jacobian_blocks_lhs(block, block_F_T_k_w, jac_block);
+        multiply_add_jacobian_blocks_rhs(rhs_block, block_F_T_k_w, measured_power, f_x_complex);
+    }
+
+    /// @brief Adds contribution of branch measurements to the G_(r, r), G_(r, c) and eta_r blocks,
+    ///  given the iteration passes through (r, c) ie. row, col
+    ///
+    /// When iterating via (row, col), have 4 cases regarding branch measurements:
+    ///      if y_type == yft
+    ///          if from_measurement,
+    ///              xi = f, mu = t, chi = row, psi = col
+    ///          if to_measurement,
+    ///              xi = t, mu = f, chi = col, psi = row
+    ///      if y_type == ytf &
+    ///          if from_measurement,
+    ///              xi = f, mu = t, chi = col, psi = row
+    ///          if to_measurement,
+    ///              xi = t, mu = f, chi = row, psi = col
+    ///
+    /// f_P_(x) = -M(U_chi, U_chi, Y_xi_xi) - M(U_chi, U_psi, Y_xi_mu)
+    /// f_Q_(x) = H(U_chi, U_chi, Y_xi_xi) + H(U_chi, U_psi, Y_xi_mu)
+    ///
+    ///
+    /// @param block G_(r, c)
+    /// @param diag_block G_(r, r)
+    /// @param rhs_block RHS(r)
+    /// @param y_xi_xi shunt admittance near to branch measurement
+    /// @param y_xi_mu admittance from the branch measurement to other bus
+    /// @param u_state Voltage state of iteration voltage state vector
+    /// @param order bool to determine if (chi, psi) = (row, col) or (col, row)
```

Comment:
Would it make sense to make this bool `order` an enum to be more descriptive? Say `row_major` and `column_major`? Or perhaps make the bool variable name more descriptive, e.g., `row_ordered`.